### PR TITLE
.github/workflow: Fix prometheus 3 tests

### PR DESCRIPTION
**Pull request by slashpai**
## Description

_Describe the big picture of your changes here to communicate to the maintainers why we should accept this pull request.
If it fixes a bug or resolves a feature request, be sure to link to that issue._

## Type of change

_What type of changes does your code introduce to the Prometheus operator? Put an `x` in the box that apply._

- [ ] `CHANGE` (fix or feature that would cause existing functionality to not work as expected)
- [ ] `FEATURE` (non-breaking change which adds functionality)
- [ ] `BUGFIX` (non-breaking change which fixes an issue)
- [ ] `ENHANCEMENT` (non-breaking change which improves existing functionality)
- [x] `NONE` (if none of the other choices apply. Example, tooling, build system, CI, docs, etc.)

## Verification

Please check the [Prometheus-Operator testing guidelines](../TESTING.md) for recommendations about automated tests.

## Changelog entry

_Please put a one-line changelog entry below. This will be copied to the changelog file during the release process._

```release-note

```

### CODE REVIEW

**Comment by slashpai** [comment]
@simonpasquier Tried running with previous commit https://github.com/prometheus-operator/prometheus-operator/actions/runs/10994565459/job/30523402743?pr=6959

**Comment by slashpai** [comment]
Need to look into following tests failing for Prometheus 3

```
 --- FAIL: TestAllNS/y/PromRemoteWriteWithTLS (0.00s)
            --- FAIL: TestAllNS/y/PromRemoteWriteWithTLS/variant-1 (43.28s)
            --- FAIL: TestAllNS/y/PromRemoteWriteWithTLS/variant-11 (43.69s)
            --- FAIL: TestAllNS/y/PromRemoteWriteWithTLS/variant-20 (44.74s)
            --- PASS: TestAllNS/y/PromRemoteWriteWithTLS/variant-19 (44.23s)
            --- PASS: TestAllNS/y/PromRemoteWriteWithTLS/variant-18 (44.22s)
            --- PASS: TestAllNS/y/PromRemoteWriteWithTLS/variant-16 (44.88s)
            --- PASS: TestAllNS/y/PromRemoteWriteWithTLS/variant-15 (44.94s)
            --- PASS: TestAllNS/y/PromRemoteWriteWithTLS/variant-17 (59.82s)
            --- PASS: TestAllNS/y/PromRemoteWriteWithTLS/variant-14 (45.38s)
            --- FAIL: TestAllNS/y/PromRemoteWriteWithTLS/variant-12 (44.09s)
            --- FAIL: TestAllNS/y/PromRemoteWriteWithTLS/variant-6 (45.05s)
            --- PASS: TestAllNS/y/PromRemoteWriteWithTLS/variant-13 (60.40s)
            --- FAIL: TestAllNS/y/PromRemoteWriteWithTLS/variant-10 (45.65s)
            --- FAIL: TestAllNS/y/PromRemoteWriteWithTLS/variant-9 (44.84s)
            --- FAIL: TestAllNS/y/PromRemoteWriteWithTLS/variant-8 (45.65s)
            --- FAIL: TestAllNS/y/PromRemoteWriteWithTLS/variant-7 (46.31s)
            --- FAIL: TestAllNS/y/PromRemoteWriteWithTLS/variant-4 (46.05s)
            --- FAIL: TestAllNS/y/PromRemoteWriteWithTLS/variant-5 (48.28s)
            --- FAIL: TestAllNS/y/PromRemoteWriteWithTLS/variant-3 (49.39s)
            --- FAIL: TestAllNS/y/PromRemoteWriteWithTLS/variant-2 (48.76s)
```

**Comment by slashpai** [comment]
As per https://www.prometheus.io/blog/2024/09/11/prometheus-3-beta/#other-breaking-changes, `remote-write-receiver` is removed in Prometheus 3, so we need to fix that as well 

https://github.com/prometheus-operator/prometheus-operator/blob/23cbc111619cbe089a04ae81463584e88a183af9/test/framework/prometheus.go#L292

File: test/framework/prometheus.go [hunk 1]
```diff
@@ -289,8 +291,11 @@ func (prwtc PromRemoteWriteTestConfig) AddRemoteWriteWithTLSToPrometheus(p *moni
 }
 
 func (f *Framework) EnableRemoteWriteReceiverWithTLS(p *monitoringv1.Prometheus) {
-	p.Spec.EnableFeatures = []monitoringv1.EnableFeature{"remote-write-receiver"}
-
+	promVersion, _ := semver.ParseTolerant(p.Spec.Version)
+	// remote-write-receiver flag is removed in Prometheus 3
+	if promVersion.Major == 2 {
+		p.Spec.EnableFeatures = []monitoringv1.EnableFeature{"remote-write-receiver"}
```
[the remark this answers is not in the document]
Comment (slashpai) [reply test/framework/prometheus.go:297]: This was already deprecated flag (but prometheus removed in v3). I am not sure why we were setting this here but removing it was failing tests. I will check

[the remark this answers is not in the document]
Comment (slashpai) [reply test/framework/prometheus.go:297]:
```
 ts=2024-09-25T06:46:05.113Z caller=dedupe.go:112 component=remote level=error remote_name=e8f1fc url=https://prometheus-test-receiver:9090/api/v1/write msg="non-recoverable error" failedSampleCount=49 failedHistogramCount=0 failedExemplarCount=0 err="server returned HTTP status 404 Not Found: remote write receiver needs to be enabled with --web.enable-remote-write-receiver\n"
```

Setting the deprecated feature flag was still working on v2 thats why it didn't error out for v3 we need to set it explicitly with the field 

```
EnableRemoteWriteReceiver bool `json:"enableRemoteWriteReceiver,omitempty"
```

File: test/framework/prometheus.go [hunk 2]
```diff
@@ -289,8 +290,7 @@ func (prwtc PromRemoteWriteTestConfig) AddRemoteWriteWithTLSToPrometheus(p *moni
 }
 
 func (f *Framework) EnableRemoteWriteReceiverWithTLS(p *monitoringv1.Prometheus) {
-	p.Spec.EnableFeatures = []monitoringv1.EnableFeature{"remote-write-receiver"}
```
Comment (slashpai) [inline test/framework/prometheus.go:292]: I updated to common field that works for both

File: test/framework/prometheus.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (slashpai) [reply test/framework/prometheus.go]: I updated to https://github.com/prometheus-operator/prometheus-operator/pull/6959/files#r1774629459 that works both tested locally for both versions it works.

I will check more issues if comes in other tests and fix separately

**Comment by slashpai** [comment]
@simonpasquier can you review? Fixed the remote write test

**Review by simonpasquier — APPROVED** [review]

**Pull request merged**